src/clarity/debug/dap/mod.rs

```diff
@@ -0,0 +1,1146 @@
+
+    pub fn handle_request(
+        &mut self,
+        env: Option<&mut Environment>,
+        context: Option<&LocalContext>,
+        seq: i64,
+        command: RequestCommand,
+    ) -> bool {
+        use debug_types::requests::RequestCommand::*;
+        let proceed = match command {
+            Initialize(arguments) => self.initialize(seq, arguments),
+            Launch(arguments) => self.launch(seq, arguments),
+            ConfigurationDone => self.configuration_done(seq),
+            SetBreakpoints(arguments) => self.setBreakpoints(seq, arguments),
+            SetExceptionBreakpoints(arguments) => self.setExceptionBreakpoints(seq, arguments),
+            Disconnect(arguments) => self.quit(seq, arguments),
+            Threads => self.threads(seq),
+            StackTrace(arguments) => self.stack_trace(seq, arguments),
+            Scopes(arguments) => self.scopes(seq, arguments),
+            Variables(arguments) => self.variables(seq, arguments),
+            StepIn(arguments) => self.step_in(seq, arguments),
+            StepOut(arguments) => self.step_out(seq, arguments),
+            Next(arguments) => self.next(seq, arguments),
+            Continue(arguments) => self.continue_(seq, arguments),
+            Pause(arguments) => self.pause(seq, arguments),
+            Evaluate(arguments) => self.evaluate(seq, arguments, env, context),
+            _ => {
+                self.send_response(Response {
+                    request_seq: seq,
+                    success: false,
+                    message: Some("unsupported request".to_string()),
+                    body: None,
+                });
+                false
+            }
+        };
+
+        proceed
+    }
+
+    pub fn handle_event(&mut self, seq: i64, event: Event) {
+        let response = Response {
+            request_seq: seq,
+            success: true,
+            message: None,
+            body: None,
+        };
+        self.send_response(response);
+    }
+
+    pub fn handle_response(&mut self, seq: i64, response: Response) {
+        let response = Response {
+            request_seq: seq,
+            success: true,
+            message: None,
+            body: None,
+        };
+        self.send_response(response);
+    }
+
+    // Request handlers
+
+    fn initialize(&mut self, seq: i64, arguments: InitializeRequestArguments) -> bool {
+        let capabilities = Capabilities {
+            supports_configuration_done_request: Some(true),
+            supports_function_breakpoints: Some(true),
+            supports_step_in_targets_request: Some(true),
+            support_terminate_debuggee: Some(true),
+            supports_loaded_sources_request: Some(true),
+            supports_data_breakpoints: Some(true),
+            supports_breakpoint_locations_request: Some(true),
+            supports_conditional_breakpoints: None,
+            supports_hit_conditional_breakpoints: None,
+            supports_evaluate_for_hovers: None,
```

Comment:
this capability would be pretty cool!

Reply:
Agreed. And it's probably not difficult to add. I'll open an issue to do it after the initial release.

Reply:
Actually, I remembered this seems to already work, even though it is not specified here. But maybe this flag enables something more.

<img width="376" alt="image" src="https://user-images.githubusercontent.com/1473715/167172315-483ab33b-5f2e-48ac-bc60-1e93aa4eee71.png">